### PR TITLE
Enum feature

### PR DESCRIPTION
**Pull request by queeksm**
Pull request to the development brand according to the Git Flow standards.
This pull request has all the enumerable methods according to the project

### CODE REVIEW

File: my_enums.rb [hunk 1]
```diff
@@ -1,70 +1,159 @@
...
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return enum = self.to_enum unless block_given?
+    my_each do |n|
+      yield(n, self.index(n))
     end
   end
```
Comment (whiz25) [inline my_enums.rb]: - [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] calls the given block once for each element in self

File: my_enums.rb [hunk 2]
```diff
@@ -1,70 +1,159 @@
...
+      end
+    else
+      return false if self.length == 0
+      my_each do |n|
+        return false if n.nil? || n == false
+      end
+      return true
     end
-    emp_arr == self
   end
```
Comment (whiz25) [inline my_enums.rb]: - [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: my_enums.rb [hunk 3]
```diff
@@ -1,70 +1,159 @@
...
+        if n.nil? || n == false
+          next
+        else
+          return true
+        end
+      end
+      return false
     end
-    tester
   end
```
Comment (whiz25) [inline my_enums.rb]: - [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: my_enums.rb [hunk 4]
```diff
@@ -1,70 +1,159 @@
...
+      end
+    else
+      return true if self.length == 0
+      my_each do |n|
+        return true if n.nil? || n == false
+      end
+      return false
     end
-    tester
   end
```
Comment (whiz25) [inline my_enums.rb]: - [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true only if none of the collection members is true
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good job :+1: 
You have implemented most of the features required, that's good. We need to however fix a few things before we can merge this PR. See my comments for suggestions.

### Martin Afani

**Comment by queeksm** [comment]
This pull request was marked as invalid will create a new one.

**Pull request closed without merging**